### PR TITLE
[MeshMoving] refactor testing

### LINKED ISSUE #3363: MPI mesh moving tests are failing
Currently the mpi mesh moving tests are failing with the following error (full debug output):

```
MeshMovingAnalysis: Using the old way to create the processes, this will be removed! 
MeshMovingAnalysis: Using the old way to create the processes, this will be removed! 
Info: attempting to delete inexisting modelpart : StructuralMeshMovingPart
Info: attempting to delete inexisting modelpart : StructuralMeshMovingPart
::[MeshSolverBase]:: Finished initialization.
E[andreas-ThinkPad-T430s:05749] *** An error occurred in MPI_Allreduce
[andreas-ThinkPad-T430s:05749] *** reported by process [2233073665,1]
[andreas-ThinkPad-T430s:05749] *** on communicator MPI_COMM_WORLD
[andreas-ThinkPad-T430s:05749] *** MPI_ERR_TRUNCATE: message truncated
[andreas-ThinkPad-T430s:05749] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
```

@philbucher: Does this result from the changes you made in the trilinos strategies? Should I try to fix it or are you planning to do more changes there?

### PR DESCRIPTION
**Pull request by philbucher**
This PR refactors the python-tests in the MeshMovingApplication

The reason is that there are many things duplicated and also some warnings were removed.
=> I defined a generic test-case, which is used by all other test-cases, meaning there is only one project-parameter-file, that will be slightly adapted depending on the respective test (e.g. select the domain size)
Another feature is that it unifies the testing btw serial and MPI (although there is still and issue which I have to figure out with @jcotela )

this will also close #3363 once I figure out why the node numbering changes in MPI, even though I also use the reorderIO in serial ...

So far for testing in MPI we use the point-output-process which works with coordinates, but this is not a nice solution for many points to be checked, here the ´from_json_check_process´ is much more suitable, unfortunately it works with Ids (reasonable choice for serial) but as I said there is still an issue bcs the numbering changes in MPI and I haven't yet figured out why

### CODE REVIEW

File: applications/MeshMovingApplication/tests/test_results/mesh_moving_rectangle_structural_similarity_2D4N_results.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"TIME": [0.2, 0.30000000000000004, 0.5], "NODE_55": {"MESH_DISPLACEMENT_X": [0.5996296467199931, -217.19207868149792, -29033349.487693418], "MESH_DISPLACEMENT_Y": [0.08072963054358105, -29.232764059679344, -3907724.5270597623], "MESH_DISPLACEMENT_Z": [0.0, 0.0, 0.0], "MESH_VELOCITY_X": [0.06595926113919924, -11.85262173224079, -1588914.8072574805], "MESH_VELOCITY_Y": [0.008880259359793917, -1.5952889305481095, -213858.9405877449], "MESH_VELOCITY_Z": [0.0, 0.0, 0.0]}, "NODE_95": {"MESH_DISPLACEMENT_X": [4.5251559830210475, -1638.0601369730375, -218970152.59532025], "MESH_DISPLACEMENT_Y": [1.5204292042307508, -552.6019807582369, -73867746.36491822], "MESH_DISPLACEMENT_Z": [0.0, 0.0, 0.0], "MESH_VELOCITY_X": [0.4977671581323153, -89.39190835614882, -11983629.993660953], "MESH_VELOCITY_Y": [0.1672472124653826, -30.15744241504727, -4042577.180357512], "MESH_VELOCITY_Z": [0.0, 0.0, 0.0]}}
```
Comment (philbucher) [inline applications/MeshMovingApplication/tests/test_results/mesh_moving_rectangle_structural_similarity_2D4N_results.json:1]: referencing for #4324 
@roigcarlo here the node-ids are used for checking the values

**Review by rubenzorrilla — COMMENTED** [review]
No complaints from me,

**Review by AndreasWinterstein — APPROVED** [review]
Looks fine for me.

**Comment by philbucher** [comment]
I am merging this now, then I can continue with the other cleanup

**Pull request merged**